src/main/java/io/github/darealturtywurty/turtychemistry/menu/FoundryMenu.java

```diff
@@ -0,0 +1,95 @@
+package io.github.darealturtywurty.turtychemistry.menu;
+
+import io.github.darealturtywurty.turtychemistry.block.entity.ClayAlloyFurnaceBlockEntity;
+import io.github.darealturtywurty.turtychemistry.block.entity.FoundryBlockEntity;
+import io.github.darealturtywurty.turtychemistry.init.BlockInit;
+import io.github.darealturtywurty.turtychemistry.init.MenuInit;
+import io.github.darealturtywurty.turtychemistry.menu.slot.SlotWithRestriction;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.*;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.ItemStackHandler;
+import net.minecraftforge.items.SlotItemHandler;
+import org.jetbrains.annotations.NotNull;
+
+public final class FoundryMenu extends AbstractContainerMenu {
+    private final ContainerLevelAccess access;
+    private final ContainerData data;
+
+    private FoundryMenu(final int id, final Inventory playerInventory, final IItemHandler slots, final BlockPos pos, final ContainerData data) {
+        super(MenuInit.FOUNDRY.get(), id);
+        this.access = ContainerLevelAccess.create(playerInventory.player.level, pos);
+        this.data = data;
+
+        final int slotSizePlus2 = 18;
+
+        // Fuel Slot
+        this.addSlot(new SlotWithRestriction(slots, 0, 56, 53, ClayAlloyFurnaceBlockEntity::isFuel));
+
+        // Input Slots
+        this.addSlot(new SlotItemHandler(slots, 1, (int) 55.5f, 17));
```

Comment:
```suggestion
        this.addSlot(new SlotItemHandler(slots, 1, 55, 17));
```
I don't know what on earth you did here. It honestly seems more like a decompiler artifact. Just be a little more careful please.

Reply:
ooof I have 0 idea how that happened, I think the code auto complete may have messed with that